hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -69,6 +71,8 @@ abstract class BaseMergeOnReadSnapshotRelation(sqlContext: SQLContext,
                                                prunedDataSchema: Option[StructType])
   extends HoodieBaseRelation(sqlContext, metaClient, optParams, userSchema, prunedDataSchema) {
 
+  fileIndex.setIncludeLogFiles(true)
```

Comment:
I feel this is not elegant and is left open and the caller can invoke anytime. 
Can we introduce a protected method called shouldIncludeLogFiles() to HoodieBaseRelation and set the default value to false. 

and we can override that in MergeOnReadSnapshotRelation or BaseMergeOnReadSnapshotRelation and set the value to true. 

we can change the constructor or HoodieFileIndex to take an additional arg called shouldIncludeLogFiles, and the callers are expected to set it when invoking the constructor

Reply:
Addressed